### ISSUE TITLE
Cannot read property 'map' of undefined in deconflictConstructorDeclaration

### ISSUE THREAD

**Issue by ggoodman** [wessberg/rollup-plugin-ts#108]
- Version: `1.3.1`
- Rollup Version: `2.23.0`
- Operating System and version (if applicable): N/A
- Node Version (if applicable): N/A
- Does it work with `tsc` (if applicable): ✅

### Reproduction

N/A (yet)

#### Debug work done so far:

1. Inserted a breakpoint in `deconflictConstructorDeclaration`.
2. Noted the value of `node`.
```js
node.kind === 162;
node.flags === 8;
typeof node.parameters === 'undefined';
node.original.kind === 162;
node.original.flags === 256;
Array.isArray(node.original.parameters) === true;
```
3. Noted the value of `node.original.getText()`.
```js
private constructor(
    readonly kind: SourceModuleDependencyKind,
    readonly spec: string,
    readonly locations: ReadonlyArray<SourceLocation>,
    readonly options: SourceModuleOptions = {}
  ) {}
```

My wild guess is that there may be something weird going on with private constructors.

### Expected Behavior

- Builds successfully

### Actual Behavior

- Throws uncaught exception

**Comment by wessberg** [comment]
Hey there. I don't really have that much to go on here. Do you have an example of some code that will throw? And/or a stack trace when the error is thrown?

**Comment by ggoodman** [comment]
@wessberg I've created a minimal reproduction here: https://github.com/ggoodman/rollup-plugin-ts-bug/tree/rollup-plugin-ts-108

The following code should be able to reproduce the issue:

```typescript
export class Singleton {
  private constructor() {}

  private static instance: Singleton;

  static getInstance() {
    if (!Singleton.instance) {
      Singleton.instance = new Singleton();
    }

    return Singleton.instance;
  }
}
```

**Comment by wessberg** [comment]
Hey there. I get a 404 when I try to go to https://github.com/ggoodman/rollup-plugin-ts-bug/tree/rollup-plugin-ts-108. Is the repo private? I tried to set up a testcase with the code example you provided instead, but that one didn't fail.

**Comment by ggoodman** [comment]
I **KEEP** doing that.. 🤦 . The repo is now public.

**Comment by ggoodman** [comment]
@wessberg dropping a mention to clarify that the repro repo should be publicly visible.

**Comment by wessberg** [comment]
Alright, this has been fixed now. This only breaks in TypeScript versions before v4, but is related to [this issue](https://github.com/microsoft/TypeScript/issues/40361) that I've just reported over at the TypeScript repo. I've added a simple null-check to avoid running into this pitfall. It has been released as part of [v1.3.4](https://github.com/wessberg/rollup-plugin-ts/releases/tag/v1.3.4)

**Comment by ggoodman** [comment]
Just took a look at your issue. I bet you're navigating a bit of a minefield of these sort of strictly-typed (except when they're not) gotchas. Amazing work!
